gtk4/src/expression.rs

```diff
@@ -0,0 +1,185 @@
+use glib::translate::*;
+use glib::{IsA, Object, Type, Value};
+
+glib::wrapper! {
+    #[derive(Debug)]
+    pub struct Expression(Shared<ffi::GtkExpression>);
+
+    match fn {
+        ref => |ptr| ffi::gtk_expression_ref(ptr),
+        unref => |ptr| ffi::gtk_expression_unref(ptr),
+        get_type => || ffi::gtk_expression_get_type(),
+    }
+}
+
+impl Expression {
+    #[doc(alias = "gtk_expression_get_value_type")]
+    pub fn get_value_type(&self) -> Type {
+        unsafe { from_glib(ffi::gtk_expression_get_value_type(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "gtk_expression_is_static")]
+    pub fn is_static(&self) -> bool {
+        unsafe { from_glib(ffi::gtk_expression_is_static(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "gtk_expression_bind")]
+    pub fn bind<T: IsA<Object>>(
+        &self,
+        target: &T,
+        property_name: &str,
+        this: Option<&T>,
+    ) -> ExpressionWatch {
+        unsafe {
+            let ret = ffi::gtk_expression_bind(
+                self.to_glib_none().0,
+                target.as_ref().to_glib_none().0,
+                property_name.to_glib_none().0,
+                this.map(|t| t.as_ref()).to_glib_none().0,
+            );
+            ExpressionWatch(*ret)
+        }
+    }
+
+    #[doc(alias = "gtk_expression_evaluate")]
+    pub fn evaluate<T: IsA<Object>>(&self, this: Option<&T>) -> Option<Value> {
+        unsafe {
+            let mut value = Value::uninitialized();
+            let ret = ffi::gtk_expression_evaluate(
+                self.to_glib_none().0,
+                this.map(|t| t.as_ref()).to_glib_none().0,
+                value.to_glib_none_mut().0,
+            );
+            if from_glib(ret) {
+                Some(value)
+            } else {
+                None
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct ExpressionWatch(ffi::GtkExpressionWatch);
```

Comment:
This one is supposed to be an opaque struct per the docs: "An opaque structure representing a watched GtkExpression."

I have no idea how to handle the exposed ref/unref methods in such case. Any suggestions?

Reply:
In what sense? You can use `glib_wrapper!` here as usual, there is just no `get_type()` function. Can't this even be autogenerated?

Reply:
>  Can't this even be autogenerated?

aha, Indeed. Although, it miss-generates the evaluate function due to missing annotations. I have kept it manual for now :) thanks!